lib/livebook/file_system.ex

```diff
@@ -28,6 +28,23 @@ defprotocol Livebook.FileSystem do
 
   @type access :: :read | :write | :read_write | :none
 
+  @doc """
+  Returns a term uniquely identifying the resource used as a file
+  system.
+  """
+  @spec resource_identifier(t()) :: term()
+  def resource_identifier(file_ystem)
+
+  @doc """
+  Checks if the file system is a resource local to its host.
+
+  A regular file system is local, because it is accessible only
+  to the host its mounted on. On the other hand, external storage
+  is global, because it can be accessed from any host.
+  """
+  @spec local?(t()) :: boolean()
+  def local?(file_system)
```

Comment:
Suggestion: let's not use a boolean function, but have a function called `type` or similar that returns an atom. This has the benefit that, if we add a new type in the future, we won't need new functions. :)

Reply:
I changed the protocol to have `type` instead, but I left `File.local?`, since it's just an abstraction layer and is convenient in this case :)